src/Features/Core/Portable/StackTraceExplorer/ParsedFrameWithFile.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.StackTraceExplorer
+{
+    internal sealed class ParsedFrameWithFile : ParsedStackFrame
+    {
+        public TextSpan FileSpan { get; }
+
+        public ParsedFrameWithFile(
+            string originalLine,
+            TextSpan classSpan,
+            TextSpan methodSpan,
+            TextSpan argsSpan,
+            TextSpan fileSpan)
+            : base(originalLine, classSpan, methodSpan, argsSpan)
+        {
+            Contract.Requires(fileSpan.Length > 0);
+            FileSpan = fileSpan;
+        }
+
+        public string GetFileText()
+        {
+            return OriginalText[FileSpan.Start..FileSpan.End];
+        }
+
+        public string GetTextBetweenTypeAndFile()
```

Comment:
> Would it make more sense to have a collection of spans available (from the base class potentially?), in order, with an indication of what type of span it is (file, class, method, text, line number etc.)

I like this idea. Kind of like enumerating "classified" text instead of providing individual values. I'll think on this more...

The spans are already exposed, and if we add a grid I think we'll still want a pure text view and swap between the two

Reply:
With the lexer/parser idea I think this becomes an even better argument to have. Instead of exposing spans it will likely expose concrete types that represent the part of the stack (fully qualified type, method, arguments, etc) and then map those to classification that we use in inlines.